### ISSUE TITLE
Autocommit warning when using isolation_level="AUTOCOMMIT"

### ISSUE THREAD

**Issue by CaselIT** [sqlalchemy/sqlalchemy#6701]
**Describe the bug**

The sa20 autocommit warning is raised even for engines created using `isolation_level="AUTOCOMMIT"`. I think it should not raise it in this case.

**To Reproduce**
Provide your [Minimal, Complete, and Verifiable](http://stackoverflow.com/help/mcve) example
here.

```py
# Insert code here

import sqlalchemy as sa

engine = sa.create_engine(
    "mssql+pyodbc://scott:tiger^5HHH@mssql_199", isolation_level="AUTOCOMMIT"
)
with engine.connect() as conn:
    conn.exec_driver_sql("DROP TABLE IF EXISTS foo")
    conn.exec_driver_sql("CREATE TABLE foo (id int primary key)")
    conn.execute(sa.text("INSERT INTO foo (id) VALUES (1)"))
```

**Error**
Provide the complete text of any errors received **including the complete
stack trace**.   If the message is a warning, run your program with the
``-Werror`` flag:   ``python -Werror myprogram.py``

```
# Copy complete stack trace and error message here, including SQL log output
if applicable.

RemovedIn20Warning: The current statement is being autocommitted using implicit autocommit, which will be removed in SQLAlchemy 2.0. Use the .begin() method of Engine or Connection in order to use an explicit transaction for DML and DDL statements. (Background on SQLAlchemy 2.0 at: http://sqlalche.me/e/b8d9)
```

**Versions.**
 - OS: -
 - Python: 3.8
 - SQLAlchemy: 1.4
 - Database: - 
 - DBAPI: - 

**Additional context**

**Have a nice day!**

**Comment by zzzeek** [comment]
this is due to the legacy "isolation_level" setting that I'm not sure people are using, which is discussed in #6342. if isolation level is set using the [prominently documented technique](https://docs.sqlalchemy.org/en/14/core/connections.html#setting-transaction-isolation-levels-including-dbapi-autocommit), the warning does not emit as there is [conditional logic to test this](https://github.com/sqlalchemy/sqlalchemy/blob/master/lib/sqlalchemy/engine/base.py#L969).  If we fix #6342 to establish the isolation level as an execution option, then this gets fixed also.   i dont like that there are two totally different systems in place for this.  docs at https://docs.sqlalchemy.org/en/14/core/engines.html?highlight=create_engine#sqlalchemy.create_engine.params.isolation_level are also not well cross-linked to the modern approach and some sections like https://docs.sqlalchemy.org/en/14/dialects/sqlite.html#transaction-isolation-level-autocommit are also out of date.

**Comment by zzzeek** [comment]
if we do #6342 and ensure that create_engine.isolation_level also sets the execution option, then this will work.  or otherwise we need to decide what we are doing with #6342.    two different systems seems like a bad idea.

**Comment by CaselIT** [comment]
I thought that they where the same option, or at least end up in the same place.

I'm all for unifying them, since it does not make sense having two things that do the same stuff.
Where would the unification look like? should `isolation_level` just be a shorthand for setting the execution option?

**Comment by zzzeek** [comment]
I mostly think ``isolation_level`` should be shorthand for the execution option.   

however it does make a subtle behavioral change within the dialect - with the old option, the connection is set to the given isolation level just once when it's created.   with the new option, it gets set upon checkout and reset upon return to the pool.   The end result is the same but there is more SQL going back and forth in the latter version.   the latter version is of course more flexible but it has a little more per-transaction overhead.

so another approach might be to unify them, so that create_engine.isolation_level is still treated like the execution option, but we also leave in place the part where it sets it upon "connect" too, so that the per-checkout version doesnt actually emit any SQL.   But i dont know if this is worth it.

The "set the value upon checkout" is necessary even for the engine-level execution option, because we document how to share a single connection pool between two different engines that have the option set differently, the second example at https://docs.sqlalchemy.org/en/14/core/connections.html#setting-transaction-isolation-levels-including-dbapi-autocommit shows this.

given all that, a *third* approach is to rename create_engine.isolation_level to create_engine.default_isolation_level, deprecate the old name, then update all the docs.   then we fix this issue anyway :)   this is the most conservative approach.    it makes the already complicated docs even more complicated unless we can really figure out an orderly way of presenting all this.

**Comment by CaselIT** [comment]
Probably option 3 is the best one, but I agree that it will make the documentation more complex.

Given that I think unifying it with the execution option would be fine.

**Comment by CaselIT** [comment]
since for v2 the autocommit logic was updated, is this still relevant?

**Comment by CaselIT** [comment]
I'll try pooking around, if we no longer have in v2 and it's still in 1.4 we can just close as will not fix

**Comment by zzzeek** [comment]
it was probably fixed in 1.4

**Comment by CaselIT** [comment]
this was indeed fixed
